src/include/mca.h

```diff
@@ -544,6 +570,66 @@ void operator/=(Matrix<T> &a, const Number &number) {}
 template <class Number, class T, class>
 void operator/=(const Number &number, Matrix<T> &a) {}
 
+template <class T>
+void transpose(Matrix<T> &a) {
+    // single mode
+    Matrix<T> output(a.shape());
```

Comment:
After transpositon, `output`'s shape shoule be `Shape{a.columns(), a.rows()}` rather than `a.shape()`.